text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,484 @@
+to writing useful queries.  Users working with existing collector and
+SDK pipeline monitoring metrics have to remember at least three metric
+names and explicitly join them via custom metric queries.  For
+example, to calculate loss rate for an SDK using traditional pipeline
+metrics,
+
+```
+LossRate_MultipleMetrics = (dropped + failed) / (dropped + failed + success)
+```
+
+On the other hand, with a uniform boolean attribute indicating success
+or failure the resulting query is simpler.
+
+```
+LossRate_SingleMetric = items{success=false} / items{success=*}
+```
+
+In a typical metric query engine, after the user has entered the one
+metric name, attribute values will be automatically surfaced in the
+user interface, allowing them to make sense of the data and
+interactively build useful queries.  On the other hand, the user who
+has to query multiple metrics has to enter each metric name
+explicitly without help from the user interface.
+
+The proposed metric instrument would be named distinctly depending on
+whether it is a collector or an SDK, to prevent accidental aggregation
+of these timeseries.  The specified counter names would be:
+
+- `otelsdk.producer.items`: count of successful and failed items of
+  telemetry produced, by signal type, by an OpenTelemetry SDK.
+- `otelcol.receiver.items`: count of successful and failed items of
+  telemetry received, by signal type, by an OpenTelemetry Collector
+  receiver component.
+- `otelcol.processor.items`: count of successful and failed items of
+  telemetry processed, by signal type, by an OpenTelemetry Collector
+  receiver component.
+- `otelcol.exporter.items`: count of successful and failed items of
+  telemetry processed, by signal type, by an OpenTelemetry Collector
+  receiver component.
+
+### Recommended conventional attributes
+
+- `otel.success` (boolean): This is true or false depending on whether the
+  component considers the outcome a success or a failure.
+- `otel.outcome` (string): This describes the outcome in a more specific
+  way than `otel.success`, with recommended values specified below.
+- `otel.signal` (string): This is the name of the signal (e.g., "logs",
+  "metrics", "traces")
+- `otel.name` (string): Name of the component in a pipeline.
+- `otel.pipeline` (string): Name of the pipeline in a collector.
+
+### Specified `otel.outcome` attribute values
+
+The `otel.outcome` attribute indicates extra information about a
+success or failure.  A set of standard conventional attribute values
+is supplied and is considered a closed set.  If these outcomes do not
+accurately explain the reason for a success or failure outcome, they
+SHOULD be extended by OpenTelemetry.
+
+For success:
+
+- `consumed`: Indicates a normal, synchronous request success case.
+  The item was consumed by the next stage of the pipeline, which
+  returned success.
+- `unsampled`: Indicates a successful drop case, due to sampling.
```

Comment:
When the [Filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md) filters out data, should it count it as `unsampled`? The name doesn't sound right in this case.